web/source/kmwosk.js

```diff
@@ -699,9 +708,15 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
 
       // TODO:  Refactor the overloading of the 'n' parameter here into separate methods.
 
-      // Test for fall back to U_xxxx key id - For this first test, we refer to Unicode values.
-      if((keyName.substr(0,2) == 'U_') && (n > osk.keyCodes['K_SPACE']) &&  !(n>127 && n<160)) { // 127 - 160 refer to Unicode control codes.
-        ch=String.fromCharCode(n);
+      // Test for fall back to U_xxxx key id - For this first test, we ignore the key code
+      // and use the keyName
+      if((keyName.substr(0,2) == 'U_')) { 
+        ch=String.fromCharCode(parseInt(keyName.substr(2,6),16));
+        if(ch < 32 || (ch > 127 && ch < 160)) {
```

Comment:
Shouldn't this read `if(ch <= 32 || (ch > 127 && ch < 160)) {`?  The original test was if `n > osk.keyCodes['K_SPACE']), where `K_SPACE` should probably be 32, the ASCII code for SPACE?